### PR TITLE
COP-12077 - Intercepting officer view

### PR DESCRIPTION
**Pull request by charles474**
# Description
This PR does the following:
- Aligns the Airpax targets list page to match the MVP design for the Intercepting frontline officer.
- Adds the RoRo targets list page (labelled as **RoRo V2** in the dropdown which allows a user to switch target lists).
- Adds the record outcome form (still requires work in terms of validation but it is adequate for now).
- Minor code changes (some refactoring, including test fixtures).

## NOTE
- The targets info page is not complete with the appropriate views. These will be added in a separate PR.
- The Record outcome form only works for options other than positive exam & negative exam.

### CODE REVIEW

File: client/public/locales/en-GB/translation.json [hunk 1]
```diff
@@ -389,7 +389,16 @@
       },
       "claimed-statement": "Currently claimed by",
       "button-next": "Record outcome",
-      "button-cancel": "Cancel"
+      "button-cancel": "Cancel",
```
Comment (charles474) [inline client/public/locales/en-GB/translation.json:392]: This bit is just for demo purposes only.

File: client/public/locales/en-GB/translation.json [hunk 2]
```diff
@@ -389,7 +389,16 @@
...
+      "record-outcome": {
+        "not-allowed": {
+          "completed": "You can not record an outcome against a completed target",
+          "not-assigned": {
+            "prefix": "You can not record an outcome for target",
+            "suffix": "Target is currently assigned to"
+          }
+        }
+      }
```
Comment (charles474) [inline client/public/locales/en-GB/translation.json:401]: This bit is just for demo purposes only.

File: client/src/pages/targets/v2/RecordOutcome.jsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+              'pages.target.record-outcome.not-allowed.not-assigned.suffix'
+            )} (${assignee})`,
+          });
+        }
+      } else if ([TARGET_STATUS.RECEIPTED, TARGET_STATUS.CAPTURED].includes(targetStatus)) {
+        setCanRecordOutcome({
+          allowed: true,
+        });
+      }
+      setStatus(COMPLETE);
```
Comment (charles474) [inline client/src/pages/targets/v2/RecordOutcome.jsx:161]: This bit is just for demo purposes only until we have a more refined error message from UCD.

File: client/src/pages/targets/TargetsListPage.jsx [hunk 4]
```diff
@@ -1,352 +1,176 @@
...
-
-  const maxResults = 20;
-
+  const [areTasksLoading, setAreTasksLoading] = useState(false);
+  const [refreshTaskList, setRefreshTaskList] = useState(false);
   // PAGINATION SETTINGS
   const [activePage, setActivePage] = useState(1);
   const [tasksToReturnIndex, setTasksToReturnIndex] = useState(0);
+  const maxResults = 20;
   const itemsPerPage = 20;
```
Comment (adam-thorpe) [inline client/src/pages/targets/TargetsListPage.jsx:54]: Do we really need all 4 of these constants? The `tasksToReturnIndex` is actually being used as the offset, which can be inferred from: `( activePage - 1 ) * itemsPerPage`, so maybe a function to handle this?

It seems the bottom two have the same functionality.

File: client/src/pages/targets/TargetsListPage.jsx [hunk 5]
```diff
@@ -1,352 +1,176 @@
...
-       */
-      setRefreshTaskList(true);
-    }
+    setRefreshTaskList(true);
   };
 
-  const handleFilterReset = (e) => {
+  const handleFilterReset = (event) => {
     setTasksToReturnIndex(0);
     setActivePage(0);
```
Comment (adam-thorpe) [inline client/src/pages/targets/TargetsListPage.jsx:146]: If the filters have been reset, shouldn't the activePage be set back to 1?

File: client/src/pages/targets/TasksTab.jsx [hunk 6]
```diff
@@ -1,17 +1,22 @@
...
 
+import { TaskSelectedTabContext } from '../../utils/TabSelectedContext';
 import TargetPagination from './__components__/TargetPagination';
-import TargetListCard from './TargetListCard';
-import { cachedTabKey } from '../../utils/constants';
+import TargetListCard from './__components__/TargetListCard';
+
+import { MovementUtil } from './utils';
 
 const TasksTab = ({
```
Comment (adam-thorpe) [inline client/src/pages/targets/TasksTab.jsx:21]: File could be named TargetsTab?

File: client/src/pages/targets/TasksTab.jsx [hunk 7]
```diff
@@ -48,37 +52,16 @@ const TasksTab = ({
...
-    const { businessKey } = selectedTask.informationSheet.parentBusinessKey;
-
+    const { informationSheet } = selectedTask;
     await axiosInstance({
       method: 'POST',
-      url: `/camunda/v1/targets/${businessKey}/captures`,
+      url: `/cop-targeting-api/v2/targets/${MovementUtil.taskId(informationSheet)}/captures`,
       cancelToken: source.token,
       data: {
         user: currentUser,
```
Comment (adam-thorpe) [inline client/src/pages/targets/TasksTab.jsx:67]: Capture has been dropped so this function can be removed (Including RoRo v2 btw)

File: client/src/pages/targets/__components__/TargetListItemOptions.jsx [hunk 8]
```diff
@@ -19,26 +19,34 @@ const TargetListItemOptions = ({
...
+      {selectedTab === TASK_STATUS.NEW && [TARGET_STATUS.ACKNOWLEDGED].includes(targetStatus) && (
+        <Button
+          id="captureTarget"
+          classModifiers="blue"
+          className="margin-right-5"
+          onClick={() => handleOnCapture()}
+        >
+          Capture
         </Button>
       )}
```
Comment (adam-thorpe) [inline client/src/pages/targets/__components__/TargetListItemOptions.jsx:47]: Also can be removed

Comment (charles474) [reply client/src/pages/targets/__components__/TargetListItemOptions.jsx:47]: This component is shared between RoRo V1, Airpax & RoRo V2 hence the conditions.

File: client/src/pages/targets/TasksTab.jsx (hunk 6 above)
Comment (charles474) [reply client/src/pages/targets/TasksTab.jsx:21]: GOod shout, I hadn't thought of that.

File: client/src/pages/targets/TargetsListPage.jsx (hunk 4 above)
Comment (charles474) [reply client/src/pages/targets/TargetsListPage.jsx:54]: Some are used for the pagination component and some are used as the filtering parameters.

File: client/src/pages/targets/TargetsListPage.jsx [hunk 9]
```diff
@@ -1,352 +1,176 @@
...
-                targetStatuses: createTargetStatusArray(),
-                movementModes: filterMovementModes,
-                candidateGroups: !checked ? groupsToUse : checked,
-              },
-            ],
+            data: [toFilterParams().filterParams],
           });
-          setTaskCount(getCount(taskCountResponse.data[0].statusCounts));
+          setTaskCount(taskCountResponse.data[0]?.groupCounts?.totalCount);
+          setGroupCounts(taskCountResponse.data[0]?.groupCounts?.groupCounts);
```
[the remark this answers is not in the document]
Comment (charles474) [reply client/src/pages/targets/TargetsListPage.jsx:108]: Good shout with this.

File: client/src/pages/targets/TasksTab.jsx (hunk 7 above)
Comment (charles474) [reply client/src/pages/targets/TasksTab.jsx:67]: Forgot about this one 😛

**Review by adam-thorpe — APPROVED** [review]

**Comment by adam-thorpe** [comment]
Might be worth waiting for api PR to land before merging

**Pull request merged**